.github/workflows/publish.yml

```diff
@@ -0,0 +1,55 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Publish
+
+on:
+  push:
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - ".github/workflows/publish.yml"
+  workflow_dispatch:
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    strategy:
+      # Publish package one by one instead of flooding the registry
+      max-parallel: 1
+      matrix:
+        # Order here is sensitive, as it will be used to determine the order of publishing
+        package:
+          - "crates/iceberg"
+          - "crates/catalog/hms"
+          - "crates/catalog/rest"
+    steps:
+      - uses: actions/checkout@v4
+      - name: Dryrun ${{ matrix.package }}
+        working-directory: ${{ matrix.package }}
+        run: cargo publish --all-features --dry-run
+
+      - name: Publish ${{ matrix.package }}
+        working-directory: ${{ matrix.package }}
+        # Only publish if it's a tag and the tag is not a pre-release
+        if: ${{ startsWith(github.ref, 'refs/tags/') && !contains(github.ref, '-') }}
+        run: cargo publish --all-features
```

Comment:
Should we consider using [cargo workspaces](https://crates.io/crates/cargo-workspaces)? The advantage is that we don't need to add package every time we have a new one.

Reply:
> The advantage is that we don't need to add package every time we have a new one.

It's easy to add. I strive to avoid introducing new tools that maintainers must use.